torchx/schedulers/kubernetes_mcad_scheduler.py

```diff
@@ -490,16 +519,25 @@ def app_to_resource(
     the provided AppDef. The resource definition can be used to launch the
     app on Kubernetes.
 
+    MCAD supports retries at the APPLICATION level. In the case of multiple TorchX Roles,
+    the AppWrapper maximum number of retries
+    count is set to the minimum of the max_retries of the roles.
     """
 
     genericitems = []
 
-    unique_app_id = cleanup_str(make_unique(app.name))
+    unique_id_size = get_unique_id_size(app.name, len(app.roles))
+    if unique_id_size == 0:
```

Comment:
I'd argue that 1 character is too small for sufficient uniqueness. 2 is at ~1.3k, 3 is ~46k

It might be better to truncate app.name rather than reduce ID length

Reply:
Thanks for the suggestion. Updated to truncate app.name and keep at least 3 unique ID characters. Commit: 1a2ad6e90169c6ad694dca38d88610e1ecfe976c